### DISCUSSION TITLE
typescript for getStaticPaths function

### DISCUSSION THREAD

**Discussion by IRediTOTO** [vercel/next.js#28396] [Help]
Hi. I am trying config `types` for `getStaticPaths`, but I get stuck
Can you give me an example for this function?

```
export const getStaticPaths = async () => {
    return {
        paths: [],
        fallback: true,
    }
}
```

**Comment by oliverloops** [comment]
Hi @IRediTOTO, one of the ways you can accomplish that is using the `GetStaticPaths` type built into Next.js, and type your paths array if you need it as you see below (in a simple case): 

```
   import { GetStaticPaths } from 'next'

   export const getStaticPaths: GetStaticPaths = async () => {  
    
    let  typedPaths: string[] = ["pathname1", "pathname2"];
    
    return {
        paths: typedPaths,
        fallback: true,
    }
 }
```

Let me know if this works.
